code/modules/antagonists/eldritch_cult/eldritch_book.dm

```diff
@@ -0,0 +1,145 @@
+/obj/item/forbidden_book
+	name = "Codex Cicatrix"
+	desc = "Book describing the secrets of the veil."
+	icon = 'icons/obj/eldritch.dmi'
+	icon_state = "book"
+	w_class = WEIGHT_CLASS_SMALL
+	///Last person that touched this
+	var/mob/living/last_user
+	///how many charges do we have?
+	var/charge = 1
+	///Where we cannot create the rune?
+	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed,/turf/open/space,/turf/open/lava))
+
+/obj/item/forbidden_book/Destroy()
+	last_user = null
+	. = ..()
+
+
+/obj/item/forbidden_book/examine(mob/user)
+	. = ..()
+	if(!IS_HERETIC(user))
+		return
+	. += "The Tome holds [charge] charges."
+	. += "Use it on the floor to create a transmutation rune, used to perform rituals."
+	. += "Hit an influence in the black part with it to gain a charge."
+	. += "Hit a transmutation rune to destroy it."
+
+/obj/item/forbidden_book/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!proximity_flag || !IS_HERETIC(user))
+		return
+	if(istype(target,/obj/effect/eldritch))
+		remove_rune(target,user)
+	if(istype(target,/obj/effect/reality_smash))
+		get_power_from_influence(target,user)
+	if(istype(target,/turf/open))
+		draw_rune(target,user)
+
+///Gives you a charge and destroys a corresponding influence
+/obj/item/forbidden_book/proc/get_power_from_influence(atom/target, mob/user)
+	var/obj/effect/reality_smash/RS = target
+	to_chat(target, "<span class='danger'>You start drawing power from influence...</span>")
+	if(do_after(user,10 SECONDS,FALSE,RS))
+		qdel(RS)
+		charge += 1
+
+///Draws a rune on a selected turf
+/obj/item/forbidden_book/proc/draw_rune(atom/target,mob/user)
+
+	for(var/turf/T in range(1,target))
+		if(is_type_in_typecache(T, blacklisted_turfs))
+			to_chat(target, "<span class='warning'>The terrain doesn't support runes!</span>")
+			return
+	var/A = get_turf(target)
+	to_chat(user, "<span class='danger'>You start drawing a rune...</span>")
+
+	if(do_after(user,30 SECONDS,A))
```

Comment:
/proc/do_after(mob/user, var/delay, needhand = 1, atom/target = null, progress = 1, datum/callback/extra_checks = null)
needs to have target=A rather than needhand=A